indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -694,6 +710,28 @@ public Response doGetLog(
     }
   }
 
+  @GET
+  @Path("/dataSources/{dataSource}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getRunningTasksByDataSource(@PathParam("dataSource") String dataSource,
+      @Context HttpServletRequest request)
+  {
+    Optional<TaskRunner> ts = taskMaster.getTaskRunner();
+    if (ts.isPresent()) {
+      Collection<? extends TaskRunnerWorkItem> runningTasks = ts.get().getRunningTasks();
+      if (runningTasks == null || runningTasks.isEmpty()) {
+        return Response.status(Response.Status.NOT_FOUND)
+            .entity("No running tasks found for the datasource : " + dataSource).build();
+      }
+      List<TaskRunnerWorkItem> taskRunnerWorkItemList = runningTasks.stream()
+          .filter(task -> dataSource.equals(task.getDataSource())).collect(Collectors.toList());
+      return Response.ok(taskRunnerWorkItemList).build();
+    } else {
+      return Response.status(Response.Status.NOT_FOUND)
```

Comment:
suggest putting this at the top if statement. It is usually easier to read like this:

```java
if (!ts.isPresent()) {
  return Response.status(Response.Status.NOT_FOUND).entity("No tasks are running").build();
}
// more stuff here without the extra indent
```